src/database/DataTypes/StocktakeBatch.js

```diff
@@ -268,6 +286,9 @@ export class StocktakeBatch extends Realm.Object {
     this.itemBatch.sellPrice = this.sellPrice;
     this.itemBatch.supplier = this.supplier;
 
+    this.itemBatch.applyVvmStatus(database, this.vaccineVialMonitorStatus);
+    this.itemBatch.applyLocation(database, this.location);
```

Comment:
when finalising a stocktake, each stocktake batch handles it's own finalising. So when this stocktake batch is finalised, it adjusts the underlying item batches vvm status and location